pkg/snmp/snmp_test.go

```
@@ -526,6 +527,8 @@ network_devices:
 	assert.Equal(t, "legacyCommunityString", legacyConfig.Community)
 	assert.Equal(t, "legacySnmpVersion", legacyConfig.Version)
 	assert.Equal(t, "127.2.0.0/30", legacyConfig.Network)
+	warnings := pkgconfigsetup.Datadog().Warnings()
+	assert.NotNil(t, warnings)
```

Could you add a method `Count()`to `pkg/config/model/warnings.go`. It should return the numbers of `errors` that the `Warnings` object has. Compare that count here instead of checking that the object is non-nil. I believe it should be zero since my initial assumption (that there was an error being swallowed by the code) seems to be wrong.

---

Added the Count() method and used it here.